### PR TITLE
Integrating sphinx to build docs and auto-push to github pages

### PR DESCRIPTION
**Pull request by pri22296**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+Please join our mailing list to discuss questions regarding the project:
+https://groups.google.com/forum/#!forum/loklak
+
+Our chat channel is on gitter here:
+https://gitter.im/fossasia/susi_server
+
+How do I install Susi: Download, Build, Run
+===========================================
+
+    **Note**: You must be logged in to Docker Cloud for the button to
```
Comment (yukiisbored) [inline README.rst]: Please use rST's note directive

**Comment by yukiisbored** [comment]
The SUSI logo is missing

**Comment by pri22296** [comment]
@yukiisbored The SUSI logo is specified in conf.py. The logo renders on locally built docs.

File: .utility/push-docs-to-gh-pages.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+git commit -m "Latest javadoc on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to gh-pages"
+git push -fq origin gh-pages > /dev/null 2>&1
+
+if [ $? -eq 0 ]; then
+    echo -e "Published Javadoc to gh-pages.\n"
+    exit 0
+else
+    echo -e "Publishing failed. Maybe the access-token was invalid or had insufficient permissions.\n"
+    exit 1
+fi
```
Comment (yukiisbored) [inline .utility/push-docs-to-gh-pages.sh]: Missing `\n` on EOL

**Comment by yukiisbored** [comment]
@pri22296 Yes, but I don't think ` images/susi-logo-original.svg' exist inside the `docs` folder

**Comment by pri22296** [comment]
@yukiisbored susi-logo-original.svg has been added under docs/_static/images in this commit. sphinx searches for images under the _static directory so /images/susi-logo-original.svg as path is enough.

**Comment by yukiisbored** [comment]
Okay, since the push script requires an SSH key, I'm gonna add it myself. Thanks.

**Comment by pri22296** [comment]
Thanks @yukiisbored

**Pull request merged**